presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitSource.java

```diff
@@ -104,7 +127,106 @@ void fail(Throwable e)
             return failedFuture(throwable.get());
         }
 
-        return future;
+        return future.thenApply(this::dynamicallyFilterSplits);
+    }
+
+    private List<ConnectorSplit> dynamicallyFilterSplits(List<ConnectorSplit> splits)
+    {
+        List<DynamicFilterDescription> dynamicFilterDescriptions = getAvailableDynamicFilterDescriptions();
+
+        TupleDomain<HiveColumnHandle> runtimeTupleDomain = dynamicFilterDescriptions.stream()
+                .map(DynamicFilterDescription::getTupleDomain)
+                .map(domain -> domain.transform(HiveColumnHandle.class::cast))
+                .reduce(TupleDomain.all(), TupleDomain::columnWiseUnion);
```

Comment:
I think it should be `TupleDomain.none()` here instead of `TupleDomain.all` as we are doing Union (OR) here.